### PR TITLE
refac(zk): refactor floor planner

### PR DESCRIPTION
**Pull request by Insun35**
- Rename `V1Pass` to `V1Layouter`.
- Add `RegionLayouter` class and replace duplicate code.
- Add `LookupTableAssigner` class and replace duplicate code.

### CODE REVIEW

**Comment by chokobole** [comment]
Could you change the type in https://github.com/kroma-network/tachyon/pull/245/commits/99ec0b6c7c734cbdf4abeee0de1e22e81e926ce0 `chore` to `build`?

**Comment by chokobole** [comment]
maybe we need to add this to `docs/development/convention.md` as well

File: tachyon/zk/plonk/circuit/floor_planner/lookup_table_assigner.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
+#ifndef TACHYON_ZK_PLONK_CIRCUIT_FLOOR_PLANNER_LOOKUP_TABLE_ASSIGNER_H_
+#define TACHYON_ZK_PLONK_CIRCUIT_FLOOR_PLANNER_LOOKUP_TABLE_ASSIGNER_H_
+
+#include <utility>
+#include <vector>
+
```
Comment (chokobole) [inline tachyon/zk/plonk/circuit/floor_planner/lookup_table_assigner.h:6]:
```suggestion
#include <optional>
#include <string_view>
```

File: tachyon/zk/plonk/circuit/floor_planner/lookup_table_assigner.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                              typename SimpleLookupTableLayouter<F>::Value>&
+        values = lookup_table_layouter.values();
+    // Check that all table columns have the same length |first_unused|,
+    // and all cells up to that length are assigned.
+    std::optional<size_t> first_unused;
+    std::vector<std::optional<size_t>> assigned_sizes = base::Map(
+        values.begin(), values.end(),
+        [](const std::pair<LookupTableColumn,
+                           typename SimpleLookupTableLayouter<F>::Value>&
+               entry) {
```
Comment (ashjeong) [inline tachyon/zk/plonk/circuit/floor_planner/lookup_table_assigner.h:47]: Shouldn't this start of this line line up with `const` on line 45?

**Review by chokobole — APPROVED** [review]
LGTM

File: tachyon/zk/plonk/circuit/floor_planner/lookup_table_assigner.h [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+namespace tachyon::zk {
+
+template <typename F>
+class LookupTableAssigner {
+ public:
+  using AssignLookupTableCallback =
+      typename Layouter<F>::AssignLookupTableCallback;
+
+  LookupTableAssigner(Assignment<F>* assignment,
+                      std::vector<LookupTableColumn>& lookup_table_columns)
```
Comment (TomTaehoonKim) [inline tachyon/zk/plonk/circuit/floor_planner/lookup_table_assigner.h:30]:
```suggestion
                      const std::vector<LookupTableColumn>& lookup_table_columns)
```

**Review by TomTaehoonKim — CHANGES_REQUESTED** [review]

File: tachyon/zk/plonk/circuit/floor_planner/lookup_table_assigner.h (hunk 3 above)
Comment (Insun35) [reply tachyon/zk/plonk/circuit/floor_planner/lookup_table_assigner.h:30]: This shouldn't be a const cuz it pushes values into it.

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by ashjeong — APPROVED** [review]
LGTM

File: tachyon/zk/plonk/circuit/floor_planner/region_layouter.h [hunk 4]
```diff
@@ -4,45 +4,53 @@
...
+#include "tachyon/zk/plonk/circuit/assignment.h"
+#include "tachyon/zk/plonk/circuit/column_key.h"
+#include "tachyon/zk/plonk/circuit/floor_planner/constant.h"
 #include "tachyon/zk/plonk/circuit/region.h"
 
 namespace tachyon::zk {
 
 template <typename F>
-class V1Region : public Region<F>::Layouter {
+class RegionLayouter : public Region<F>::Layouter {
```
Comment (dongchangYoo) [inline tachyon/zk/plonk/circuit/floor_planner/region_layouter.h:23]: If `RegionLayouter` is the only one, an abstract class named `Region::Layouter` is unnecessary, and if not, the name `RegionLayouter` should be more specialized.

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Pull request merged**